liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -170,4 +172,16 @@ private void connect() {
   private void updateRequestPriority(LocationRequest request) {
     REQUEST_PRIORITY.get(priority).update(request);
   }
+
+  private PendingIntent getPendingIntent() {
+    Context context = this.context.get();
+    if (context != null) {
+      SdkChecker sdkChecker = new SdkChecker();
+      LocationPendingIntentProvider provider =
+        new LocationPendingIntentProvider(context, sdkChecker, locationListeners);
+      return provider.buildPendingIntent();
+    } else {
+      throw new IllegalStateException(CONTEXT_NULL);
```

Comment:
On second thought here we can build the `PendingIntent` in the constructor I believe, so we don't have to worry about the `WeakReference` being `null`